### PR TITLE
Quick Steps to creating SP credentials on Azure.

### PR DESCRIPTION
**Pull request by deekoder**

### CODE REVIEW

File: docs/azure_service_principal.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+## Create a Service Principal Account
+Pick any one of the two different ways to create a Service Principal.
+ - Azure Portal
+ - Azure CLI
+ 
+## Using Azure Portal:
+ - Login to your Azure account on https://portal.azure.com
+ - Navigate to `User setting` under `Azure Active Directory`.
```
Comment (vadmeste) [inline docs/azure_service_principal.md]: s/setting/settings

File: docs/azure_service_principal.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ - Azure Portal
+ - Azure CLI
+ 
+## Using Azure Portal:
+ - Login to your Azure account on https://portal.azure.com
+ - Navigate to `User Settings` under `Azure Active Directory`.
+ - Ensure that user can register applications, ONLY an admin user can register applications.
+ - Create an `Azure Active Directory application` [(Learn more)](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal?view=azure-cli-latest).
+ - Retrieve `application ID` and `authentication key`
+ - Assign the newly created application to a role in order to access Azure resources.
```
Comment (vadmeste) [inline docs/azure_service_principal.md]: I think it is better to say here: Assign the contributor role to the created service principal, over all storage accounts that are needed to get accessed by the application.

File: docs/azure_service_principal.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ 
+ If programmatic login is required, remember to retrieve `tenant ID` since you will need to pass this during authentication.[(Learn more)](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal?view=azure-cli-latest).
```
Comment (vadmeste) [inline docs/azure_service_principal.md]: This line is not needed.

File: docs/azure_service_principal.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  Retrying role assignment creation: 3/36
+  {
+    "appId": "04e4c5c8-4c77-4147-b1d9-00a0226be5d3",
+    "displayName": "minio-mapp",
+    "name": "http://minio-mapp",
+    "password": "XXXXXXXXX",
+    "tenant": "XXxxXXxx-XXXx-XXXx-XXXx-xxxXXXXXxXxx"
+  }
+  ```
+ - Default option for the role that the new application will take in Azure is "Contributor", this provides more privilages than usual and if required can be restricted to a role with lesser permission, such as a "Reader" role.
```
Comment (vadmeste) [inline docs/azure_service_principal.md]: s/privilages/privileges

Comment (vadmeste) [reply docs/azure_service_principal.md]: I think this paragraph is also not needed if we want a short documentation

File: docs/azure_service_principal.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    "displayName": "minio-mapp",
+    "name": "http://minio-mapp",
+    "password": "XXXXXXXXX",
+    "tenant": "XXxxXXxx-XXXx-XXXx-XXXx-xxxXXXXXxXxx"
+  }
+  ```
+ - Default option for the role that the new application will take in Azure is "Contributor", this provides more privilages than usual and if required can be restricted to a role with lesser permission, such as a "Reader" role. 
+
+```
+$ az role assignment create --assignee 04e4c5c8-4c77-4147-b1d9-00a0226be5d3 --role Reader
```
Comment (vadmeste) [inline docs/azure_service_principal.md]: s/Reader/Contributor/

File: docs/azure_service_principal.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      "updatedOn": "2018-01-18T04:18:03.5317841Z"
+    },
+    "principalId": "4ef772e9-bbe2-4ad8-8e9a-a97dd0084169",
+    "roleDefinitionId": "/subscriptions/XXxxXXxx-XXXx-XXXx-XXXx-xxxXXXXXxXxx/providers/Microsoft.Authorization/roleDefinitions/acdd72a7-3385-48ef-bd42-f606fba81ae7",
+    "scope": "/subscriptions/XXxxXXxx-XXXx-XXXx-XXXx-xxxXXXXXxXxx"
+  },
+  "type": "Microsoft.Authorization/roleAssignments"
+}
+
+$ az role assignment delete --assignee 04e4c5c8-4c77-4147-b1d9-00a0226be5d3 --role Contributor
```
Comment (vadmeste) [inline docs/azure_service_principal.md]: Remove this

File: docs/azure_service_principal.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+$ az role assignment list   --assignee 04e4c5c8-4c77-4147-b1d9-00a0226be5d3
```
Comment (vadmeste) [inline docs/azure_service_principal.md]: Remove this

File: docs/azure_service_principal.md (hunk 2 above)
Comment (deekoder) [reply docs/azure_service_principal.md]: ok

File: docs/azure_service_principal.md (hunk 5 above)
Comment (deekoder) [reply docs/azure_service_principal.md]: ack

**Pull request merged**